### PR TITLE
Date filters exclude boundaries

### PR DESCRIPTION
**Pull request by giorgiosironi**
Excluding because the variables were mixed up and were setting 23:59:59 for the start date and 00:00:00 for the end date.

### CODE REVIEW

File: src/Search/Workflow/ResearchArticleWorkflow.php [hunk 1]
```diff
@@ -132,7 +132,7 @@ public function index(ArticleVersion $article) : array
             'value' => json_encode($articleObject->dataSets ?? '[]'),
         ];
 
-        $sortDate = $article->getStatusDate() ? $article->getStatusDate() : $article->getPublishedDate();
+        $sortDate = $article->getStatusDate();
```
Comment (thewilkybarkid) [inline src/Search/Workflow/ResearchArticleWorkflow.php]: Makes sense to remove. If there's no status date, there won't be a published date either.

File: tests/src/Search/Web/DateRangeTest.php [hunk 2]
```diff
@@ -34,13 +39,18 @@ public function test_date_range_end_only()
...
 
         $this->jsonRequest('GET', '/search', ['end-date' => '2016-10-01']);
         $response = $this->getJsonResponse();
-        $this->assertEquals($response->total, 0);
+        $this->assertEquals(0, $response->total);
+
+        // boundary
+        $this->jsonRequest('GET', '/search', ['end-date' => '2016-12-05']);
+        $response = $this->getJsonResponse();
+        $this->assertEquals(1, $response->total, 'The date upper boundary is being excluded');
```
Comment (thewilkybarkid) [inline tests/src/Search/Web/DateRangeTest.php]: Could we list the IDs on the included results? Would make reading the tests easier.

File: tests/src/Search/Web/DateRangeTest.php [hunk 3]
```diff
@@ -16,13 +18,18 @@ public function test_date_range_start_only()
...
         $response = $this->getJsonResponse();
-        $this->assertEquals($response->total, 2);
+        $this->assertIds(['15275', '15276'], $response);
 
         $this->jsonRequest('GET', '/search', ['start-date' => '2016-11-01']);
         $response = $this->getJsonResponse();
-        $this->assertEquals($response->total, 3);
+        $this->assertIds(['15275', '15276', '19662'], $response);
+
+        // boundary is included
```
Comment (thewilkybarkid) [inline tests/src/Search/Web/DateRangeTest.php]: The previous assertion could be for `2016-12-05` instead?

Comment (giorgiosironi) [reply tests/src/Search/Web/DateRangeTest.php]: It's still two different cases: the previous one is a filter that allows all results, while the following is a filter that falls on one of the item dates. Not worth it to go property-based though...

Comment (thewilkybarkid) [reply tests/src/Search/Web/DateRangeTest.php]: Ok; 19662 has a time (ie it's not `00:00:00`), so either use that here or change 15276 to also have a time.

Comment (giorgiosironi) [reply tests/src/Search/Web/DateRangeTest.php]: The API only allows days, not times, so changing 15276

**Pull request merged**